agent/command/git.go

```diff
@@ -719,32 +719,34 @@ func (c *gitFetchProject) fetchModuleSource(ctx context.Context,
 			c.logModuleRevision(logger, revision, moduleName, "branch field in config file")
 		}
 	}
-	var owner, repo string
-	owner, repo, err = thirdparty.ParseGitUrl(module.Repo)
-	if err != nil {
-		return err
-	}
 
 	opts := cloneOpts{
-		location: module.Repo,
-		owner:    owner,
-		repo:     repo,
 		branch:   "",
 		dir:      moduleBase,
 		method:   cloneMethod,
+		location: module.Repo,
 	}
+
 	// If the module repo is using the deprecated ssh cloning method, extract the owner
-	// and repo from the string and construct a https cloning link manually.
+	// and repo from the string and save it to ops so that the an https cloning link
+	// can be constructed manually by opts.setLocation.
 	// This is a temporary workaround which will be removed once users have switched over.
-	if strings.Contains(opts.location, "git@github.com:") {
+	owner, repo, err := module.GetOwnerAndRepo()
+	if err != nil {
+		return errors.Wrapf(err, "getting module owner and repo '%s'", module.Name)
+	}
+
+	opts.owner = owner
+	opts.repo = repo
+	if strings.Contains(module.Repo, "git@github.com:") {
```

Comment:
Hmm, maybe we should change that L742 to a warning so it stands out more? Perhaps we can either mention to go to the docs, or provide a link to the docs you've written here as well

Reply:
Changed to a warning. I don't know if it's a good idea to link to docs in task logs though, I don't think we have a precedent for that. I'm hoping that users who are confused will know to look for docs or reach out in users.

Reply:
Fair enough, SGTM